currency/rates_test.go

```diff
@@ -50,33 +51,39 @@ func TestUnMarshallRates(t *testing.T) {
 			expectedError: nil,
 		},
 		{
-			desc: "Valid JSON field defines a conversions map with repeated entries, expect error",
+			desc: "Valid JSON field defines a conversions map with repeated entries, last one wins",
 			ratesJSON: `{
 				"conversions":{
 					"USD":{
 						"GBP":0.7662523901,
 						"MXN":20.00
 					},
 					"USD":{
-						"GBP":0.7662523901
-					},
+						"GBP":0.4815162342
+					}
 				}
 			}`,
-			expectedRates: Rates{},
-			expectsError:  true,
-			expectedError: errors.New("invalid character '}' looking for beginning of object key string"),
+			expectedRates: Rates{
+				Conversions: map[string]map[string]float64{
+					"USD": {
+						"GBP": 0.4815162342,
+					},
+				},
+			},
+			expectsError:  false,
+			expectedError: nil,
 		},
 	}
 
 	for _, tc := range testCases {
 		// Execute:
 		updatedRates := Rates{}
-		err := json.Unmarshal([]byte(tc.ratesJSON), &updatedRates)
+		err := jsonutil.UnmarshalValid([]byte(tc.ratesJSON), &updatedRates)
```

Comment:
Updated all tests to use UnmarshalValid because it seemed like we want to validate in tests. Does that check out?